pkg/broker/binding_operations.go

```diff
@@ -81,8 +84,36 @@ func (b Broker) Bind(ctx context.Context, instanceID string, bindingID string, d
 		}, nil
 	}
 
-	err = CompareAndReturnAppropiateResponseCode(resultingUser, user)
-	return
+	if compareBindings(user, resultingUser) {
+		return
+	}
+	return brokerapi.Binding{}, apiresponses.NewFailureResponse(errors.New("Bindings differ in their attributes"), http.StatusConflict, "")
+}
+
+func compareBindings(localUser *atlas.User, remoteUser *atlas.User) bool {
+	//Convert to maps
+	var remoteUserMap map[string]interface{}
+	var localUserMap map[string]interface{}
+
+	inrec, err := json.Marshal(remoteUser)
```

Comment:
Why can't we compare the structs directly? Why do we need to convert to maps?

What does `inrec` mean?

Reply:
We can, I left it like this seeing that it's simpler to just add more else if statements. I guess for this one we can compare them directly. Will update them.